frontend/interfaces/host.ts

```diff
@@ -203,8 +203,10 @@ export interface IHost {
   users: IHostUser[];
   device_users?: IDeviceUser[];
   munki?: IMunkiData;
-  mdm_enrollment_status: MdmStatus;
-  mdm_server_url: string;
+  mdm: {
+    enrollment_status: MdmStatus;
+    server_url: string;
+  };
```

Comment:
@mna @fleetdm/frontend  Does this mean that we are expecting either an empty object or an object with all empty values when mdm is not enabled? We had a @fleetdm/frontend discussion about this case last week, and I believe decided we'd prefer null or undefined, in which case this should be `mdm?: {...}`

Reply:
I'm not sure what the frontend implications are (I'll let y'all take care of what needs to happen there) but currently on the backend, the API guarantees that the `mdm` object will be present, but the `enrollment_status` and `server_url` fields may be null (and will be rendered as JSON null, not absent). If a different behaviour is required from the backend, let me know (e.g. do not return instead of return null, or do not return the whole `mdm` object, ...).

Reply:
@mna we want `mdm: null` when it is disabled, and in general for empty states from the API. Thank you!

Reply:
@jacobshandling, @mna in this case, I think it would be better to have
```
mdm: {
server_url: null
enrollment_status: null
}
```

rather than
```
mdm: null
```

Reply:
My thinking here is that the pattern would be safer in terms of helping guard against an API consumer failing to guard against the null-ish object (https://github.com/fleetdm/fleet/pull/9505#discussion_r1087266943)

Reply:
Agreed, thanks for the suggestion @gillespi314! 

@mna Let's go with the structure @gillespi314 outlined above. Thanks!

Reply:
In this case this is already how it would be returned, so no changes necessary.